R/DBConnection.R

```diff
@@ -106,6 +106,41 @@ setGeneric("dbSendQuery",
   valueClass = "DBIResult"
 )
 
+#' @export
+dbBreak <- structure(list(), class = "dbiAbort")
```

Comment:
`dbWithTransaction()` is another place where we may want to be able to break out cleanly (and roll back) (#115). I'd like to implement only one way of breaking out for both chunked queries and transactions, and whatever may come. In `dbFetchChunkedQuery()` there is an isolated function where the user can `return()`, but in `dbWithTransaction()` there's just a code block, and using a sentinel value seems difficult in many cases -- the condition system seems to be suited well for this situation.

Is everybody okay with

```r
dbBreak <- function() {
  signalCondition(
    structure(
      list(message = "Aborting DBI processing", call = NULL),
      class = c("dbiAbort", "condition")))
  stop("Invalid usage of dbBreak().", call. = FALSE)
}
```

and corresponding code in `dbFetchChunkedQuery()`?